tests/integration/cleanup_test.go

```diff
@@ -227,18 +227,60 @@ func (s *ShareCreateDeleteSuite) TestCreateAndDelete() {
 
 	s.T().Log("removing prerequisite resources")
 	deleteFromFiles(ctx, require, s.tc, s.commonSources)
-	time.Sleep(waitForClearTime)
-
-	rs2 := s.getCurrentResources()
-	require.Equal(len(rs2.pods.Items), len(existing.pods.Items))
-	require.Equal(len(rs2.configMaps.Items), len(existing.configMaps.Items))
-	require.Equal(len(rs2.secrets.Items), len(existing.secrets.Items))
-	require.Equal(len(rs2.services.Items), len(existing.services.Items))
-	require.Equal(len(rs2.pvcs.Items), len(existing.pvcs.Items))
-	require.Equal(
-		len(rs2.deployments.Items), len(existing.deployments.Items))
-	require.Equal(
-		len(rs2.statefulSets.Items), len(existing.statefulSets.Items))
+
+	s.requireResourcesDone(&existing)
+}
+
+func (s *ShareCreateDeleteSuite) requireResourcesDone(base *resourceSnapshot) {
+	allDone := false
+	for i := 0; i < 50 && !allDone; i++ {
```

Comment:
Can you please write this in terms of either a [require.Eventually](https://pkg.go.dev/github.com/stretchr/testify/require#Eventually) or our home grown [TryUntil](https://github.com/samba-in-kubernetes/samba-operator/blob/e27ebc6a72b5a92ef74c8c110d283300320b33aa/tests/utils/poll/poll.go#L70) ?

A purely time-based (as opposed to counter) timeout is far easier for a human in the future to reason about and modify IMO.

Reply:
Good point -- I should have do so in the first place. Will fix.